src/emulator/auth/state.ts

```diff
@@ -643,6 +664,69 @@ export class TenantProjectState extends ProjectState {
   get usageMode() {
     return this.parentProject.usageMode;
   }
+
+  get tenantConfig() {
+    return this._tenantConfig;
+  }
+
+  delete(): void {
+    this.parentProject.deleteTenant(this.tenantId);
+  }
+
+  updateTenant(update: Partial<Tenant>, updateMask: string | undefined): Tenant {
+    // Empty masks indicate a full update
+    if (!updateMask) {
```

Comment:
Hm.. Not sure if I know what you mean in terms of moving `updateMask` logic out to `operations.ts` since this update is all closely tied to the mask. Marking as resolve, but feel free to follow up / reopen if you have further thoughts about it